src/core/thread/mesh_forwarder.cpp

```diff
@@ -2113,4 +2116,19 @@ void MeshForwarder::HandleDataRequest(const Mac::Address &aMacSource, const Thre
     {}
 }
 
+void MeshForwarder::HandleDataPollTimeout(void *aContext)
+{
+    static_cast<MeshForwarder *>(aContext)->HandleDataPollTimeout();
+}
+
+void MeshForwarder::HandleDataPollTimeout(void)
+{
+    mBacktoBackPollTimeoutCounter++;
+
+    if (mBacktoBackPollTimeoutCounter < kMaxBacktoBackPollTimeouts)
+    {
+        SendMacDataRequest();
+    }
```

Comment:
Should we set `mBacktoBackPollTimeoutCounter` back to zero here?  Or once we get into this state, we never want to do quick retransmits until we successfully receive a data message?